animation_nodes/id_keys/data_types/integer_type.py

```diff
@@ -169,3 +182,135 @@ def getObjectLocation(self, object):
             p1 = Vector(object.bound_box[0])
             p2 = Vector(object.bound_box[6])
             return object.matrix_world * ((p1 + p2) / 2)
+        
+def getIDKeyIntegerName(name):
+    IDKeyName="AN*Integer*%s" % name
+    return IDKeyName
+
+def getObjectsIDKeys(idKeyName):
+    object_list=[]
+    for obj in bpy.context.selected_objects:
+        try:
+            obj[idKeyName]
+            object_list.append(obj)
+        except KeyError: pass
+    sorted_list = sorted(object_list, key=lambda obj: obj[idKeyName])
+    return sorted_list
+
+class IDKeysIntegerOffset(bpy.types.Operator):
+    bl_idname = "an.id_keys_integer_offset"
+    bl_label = "ID Keys Offset"
+    bl_description = "Offset ID Keys of selected Objects."
+
+    idKeyName: StringProperty()
+    offset_type: EnumProperty(
+        name="Type",
+        default='ADD',
+        items=(
+            ('ADD', "Add",
+             "Add number to every selected object which have Index key.\n"
+             "Before :\n"
+             "0, 1, 2, 3\n"
+             "After Offset 1 :\n"
+             "1, 2, 3, 4"),
+            ('RANDOMIZE', "Randomize",
+             "Randomize Index key for all selected objects which have Index Key.\n"
+             "Before :\n"
+             "0, 1, 2, 3\n"
+             "After :\n"
+             "1, 3, 0, 2"),
+            ('REVERT', "Revert",
+             "Revert Index for every selected object which have Index Key.\n"
+             "Before :\n"
+             "0, 1, 2, 3\n"
+             "After :\n"
+             "3, 2, 1, 0"),
+            ('FILL', "Fill Gaps",
+             "Fill Gaps in a serie of Index.\n"
+             "Before :\n"
+             "0, 1, 3, 4\n"
+             "After :\n"
+             "0, 1, 2, 3"),
```

Comment:
For multi-line descriptions use multi-line strings with the `textwrap.dedent` function.

Reply:
all right, i don't know about it, going to search for it !